src/mscorlib/src/System/Runtime/CompilerServices/ReadOnlyAttribute.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All, Inherited = false)]
```

Comment:
Maybe mark type with `EditorBrowsableAttribute(EditorBrowsableState.Never)`?
On the other hand, I think the entire CompilerServices namespace is already meant for compiler consumption (but not general consumption), so may be superfluous.